lib/install/validate/storage.go

```diff
@@ -175,3 +177,34 @@ func (v *Validator) suggestDatastore(path string, label string, flag string) {
 		}
 	}
 }
+
+func parseScratchSize(input string) int {
+
+	sizeFormat := regexp.MustCompile(`(\d+)([TMGK])`)
+	var matches []string
+	if matches = sizeFormat.FindStringSubmatch(input); len(matches) != 3 {
+		log.Errorln("Incorrect size format provided for default image size. Should be an integer followed by a unit, such as 10T, 10G, 10M, or 10K")
+		return -1
+	}
+
+	unit := matches[2]
+	var multiplier int
+	switch unit {
+	case "M":
```

Comment:
Didn't know it existed. In any case, I was half expecting something like this to be brought up to me in review, so I didn't spend long throwing together this parser. I'll strip it out & use the library -- thanks @fdawg4l